### PR TITLE
Closes #154

### LINKED ISSUE #154: Create dataset loader for Bio-SimLex
- **Name:** Bio-SimLex
- **Description:** Noun pairs with similarity scores
- **Task:** Semantic Similarity
- **Paper:** https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2039-z
- **Data:** https://github.com/cambridgeltl/bio-simverb/tree/master/wvlib/word-similarities/bio-simlex
- **License:** ?
- **Motivation:** Evaluation

### LINKED ISSUE #155: Create dataset loader for UMNSRS Reference Standard
- **Name:** UMNSRS Reference Standard
- **Description:** UMNSRS, developed by Pakhomov, et al., consists of 725 clinical term pairs whose semantic similarity and relatedness. 
- **Task:** Semantic Similarity
- **Paper:** https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3041430/
- **Data:** https://nlp.cs.vcu.edu/data.html#umnsrs
- **License:** ?
- **Motivation:** Evaluation

### PR DESCRIPTION
**Pull request by tanmaylaud**
Name: Bio-SimLex
Description: Noun pairs with similarity scores
Task: Semantic Similarity
Paper: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2039-z
Data: https://github.com/cambridgeltl/bio-simverb/tree/master/wvlib/word-similarities/bio-simlex
License: ?
Motivation: Evaluation
#154 
### Checkbox

- [X] Confirm that this PR is linked to the dataset issue.
- [X] Create the dataloader script `biodatasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [X] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_BIGBIO_VERSION` variables.
- [X] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [X] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [X] Confirm dataloader script works with `datasets.load_dataset` function.
- [X] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio biodatasets/my_dataset/my_dataset.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: biodatasets/bio_simlex/bio_simlex.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    # You will be able to load the "source" or "bigbio" configurations with
+    # ds_source = datasets.load_dataset('my_dataset', name='source')
+    # ds_bigbio = datasets.load_dataset('my_dataset', name='bigbio')
+
+    # For local datasets you can make use of the `data_dir` and `data_files` kwargs
+    # https://huggingface.co/docs/datasets/add_dataset.html#downloading-data-files-and-organizing-splits
+    # ds_source = datasets.load_dataset('my_dataset', name='source', data_dir="/path/to/data/files")
+    # ds_bigbio = datasets.load_dataset('my_dataset', name='bigbio', data_dir="/path/to/data/files")
```
Comment (galtay) [inline biodatasets/bio_simlex/bio_simlex.py:96]: can we remove these comments (line 89-96)? they are generic templates and wont apply for this particular dataset.

File: biodatasets/bio_simlex/bio_simlex.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+                        "score": score,
+                    }
+
+                elif self.config.schema == "bigbio_pairs":
+                    yield id_, {
+                        "id": str(id_),
+                        "document_id": "NULL",
+                        "text_1": word1,
+                        "text_2": word2,
+                        "label": "0" if float(score) < 5 else "1",
```
Comment (galtay) [inline biodatasets/bio_simlex/bio_simlex.py:174]: discussed with @debajyotidatta and @jason-fries , for now I think you can leave the "label" as the string representation of the number without thresholding. can you also put a comment in about what the range is please (assuming 0-9?)

**Review by galtay — CHANGES_REQUESTED** [review]

File: biodatasets/bio_simlex/bio_simlex.py (hunk 2 above)
Comment (tanmaylaud) [reply biodatasets/bio_simlex/bio_simlex.py]: done

File: biodatasets/bio_simlex/bio_simlex.py (hunk 1 above)
Comment (tanmaylaud) [reply biodatasets/bio_simlex/bio_simlex.py]: done

File: biodatasets/bio_simlex/bio_simlex.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+_SUPPORTED_TASKS = [Tasks.SEMANTIC_SIMILARITY]
+
+_SOURCE_VERSION = "1.0.0"
+
+_BIGBIO_VERSION = "1.0.0"
+
+
+class BioSimlexDataset(datasets.GeneratorBasedBuilder):
+    """Bio-SimLex enables intrinsic evaluation of word representations. Config schema as source gives score between
+    0-10 for pairs of words. Config schema as bigbio gives label 0 and 1 for pairs of words."""
```
Comment (galtay) [inline biodatasets/bio_simlex/bio_simlex.py:84]: Can we update this to match the implementation for bigbio please.

File: biodatasets/bio_simlex/bio_simlex.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+                        "text_1": word1,
+                        "text_2": word2,
+                        "label": str(score),
+                    }
+
+
+# This allows you to run your dataloader with `python bio_simlex.py` during development
+# TODO: Remove this before making your PR
+if __name__ == "__main__":
+    datasets.load_dataset(__file__)
```
Comment (galtay) [inline biodatasets/bio_simlex/bio_simlex.py:172]: remove please.

**Review by galtay — CHANGES_REQUESTED** [review]

File: biodatasets/bio_simlex/bio_simlex.py (hunk 4 above)
Comment (hakunanatasha) [reply biodatasets/bio_simlex/bio_simlex.py]: @galtay @tanmaylaud I manually removed the dataset checker above; if you're still treating this as a draft PR please feel free to replace it but remove it before making the final PR.

**Review by galtay — APPROVED** [review]

**Comment by galtay** [comment]
thanks for the help on this @hakunanatasha and nice work @tanmaylaud !

**Pull request merged**